jenkins/bootstrap.py

```diff
@@ -267,7 +315,7 @@ def finish(gsutil, paths, success, artifacts, build, version, repo):
         'passed': bool(success),
         'metadata': metadata(repo, artifacts),
     }
-    gsutil.upload_json(paths.finished, data)
+    gsutil.upload_json(paths.finished, data)  # Raise if this fails
```

Comment:
Hmm... we definitely want to still upload the build log in this situation, but I also want to fail builds that fail to upload json... I'm going to try handling this outside of this function, what do you think?